src/plugins/discover/public/components/discover_grid/discover_grid.tsx

```diff
@@ -432,14 +442,65 @@ export const DiscoverGrid = ({
     [dataView, onFieldEdited, services.dataViewFieldEditor]
   );
 
+  const visibleColumns = useMemo(
+    () => getVisibleColumns(displayedColumns, dataView, showTimeCol) as string[],
+    [dataView, displayedColumns, showTimeCol]
+  );
+
+  const getCellValue = useCallback<UseDataGridColumnsCellActionsProps['getCellValue']>(
+    (fieldName, rowIndex): CellActionValue => {
+      const value = displayedRows[rowIndex % displayedRows.length].flattened[fieldName] as
+        | CellActionValue
+        | number
+        | number[];
+      if (value == null) {
+        return value;
+      }
+      if (Array.isArray(value)) {
+        return value.map((val) => val.toString());
+      }
+      return value.toString();
```

Comment:
This value format is what security cell actions expect, right? What if other consumers of cell actions integration would like to register their actions and have access to the original field values? I think the transformation should not be a part of Discover code.

Reply:
Actually, this is the format that the CellActions framework expects, we tried to make it as generic as possible, but I admit we couldn't help but have some Security Solution bias.
Nevertheless, this is the same problem we discussed [here](https://github.com/elastic/kibana/pull/157201#discussion_r1200457347)
If other consumers of cell actions integration would like to register their actions and receive `number | number[]` value type we would have to refactor all the actions registered currently.
(I would not consider adding the `_source` type because is too specific to Discover.)

Reply:
actions in Security:
https://github.com/elastic/kibana/tree/bb495c2d5088456208ca26980d7c4c122f2743d9/x-pack/plugins/security_solution/public/actions

generic actions in the package:
https://github.com/elastic/kibana/tree/bb495c2d5088456208ca26980d7c4c122f2743d9/packages/kbn-cell-actions/src/actions

I have been testing numeric values with all the actions we have and everything works well, elastic does not care about quoting numbers.

![string_numbers](https://github.com/elastic/kibana/assets/17747913/ac157b3e-292b-4733-96b9-fa995d83fc79)
![string_numbers_2](https://github.com/elastic/kibana/assets/17747913/0321028a-4d05-4736-9aca-ed355993c7f2)

I don't think we should refactor everything now, we haven't got any problems so far using this approach. If we find a use case for number type is needed we'll add support for it, I don't think we should worry about it for now.

Reply:
@jughosta Ironically enough, while testing it I found a bug with an action.
I created a separate issue here https://github.com/elastic/kibana/issues/159298 to do the refactor and add number support.
If you want we can merge this PR as it is, and I will add a checkpoint in the issue I just created to come to Discover and remove the string casting I added.
Does it sound good?